File: app/src/main/java/com/commit451/gitlab/navigation/BrowserFallback.java
package com.commit451.gitlab.navigation;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.widget.Toast;

import com.commit451.gitlab.R;
import com.novoda.simplechromecustomtabs.navigation.NavigationFallback;

import java.lang.ref.WeakReference;

/**
 * A fallback to open the url in the browser
 */
public class BrowserFallback implements NavigationFallback {

    private WeakReference<Context> context;

    public BrowserFallback(Context context) {
        this.context = new WeakReference<>(context);
    }

    @Override
    public void onFallbackNavigateTo(Uri url) {
        Intent i = new Intent(Intent.ACTION_VIEW);
        i.setData(url);
        Context context = this.context.get();
        if (context == null) {
            return;
        }
        try {
            context.startActivity(i);
        } catch (Exception e) {
            Toast.makeText(context, R.string.error_no_browser, Toast.LENGTH_SHORT)
                    .show();
        }
    }
}

File: app/src/main/java/com/commit451/gitlab/navigation/BrowserFallback.kt
package com.commit451.gitlab.navigation

import android.content.Context
import android.net.Uri
import com.commit451.gitlab.util.IntentUtil
import com.novoda.simplechromecustomtabs.navigation.NavigationFallback
import java.lang.ref.WeakReference

/**
 * A fallback to open the url in the browser
 */
class BrowserFallback(context: Context) : NavigationFallback {

    private val context: WeakReference<Context> = WeakReference(context)

    override fun onFallbackNavigateTo(url: Uri) {
        val context = this.context.get() ?: return
        IntentUtil.openBrowser(context, url.toString())
    }
}